contracts/src/steel/OpSteel.sol

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.9;
+
+import {Encoding, Steel} from "./Steel.sol";
+
+abstract contract OpCommitmentValidator {
```

Comment:
Why make this an `abstract` contract? If concrete, it would still be possible to inherit from, while also being possible to deploy and call as an external dependency. Although they are `internal` now, the validation functions seem like they'd be reasonable to make `public`.

Reply:
I like the idea of encouraging people to inherit the `OpCommitmentValidator`, as it would be closer to the library we already have for steel.
I am not sure if there is a real demand to make this contract publicly available, as the actual validation code is rather small and simple, and calling an external contract would increase the cost of gas.